lua/lspconfig/util.lua

```diff
@@ -124,25 +130,41 @@ M.path = (function()
 
   local dirname
   do
-    local strip_dir_pat = path_sep .. '([^' .. path_sep .. ']+)$'
-    local strip_sep_pat = path_sep .. '$'
     dirname = function(path)
+      local strip_dir_pat = path_sep() .. '([^' .. path_sep() .. ']+)$'
+      local strip_sep_pat = path_sep() .. '$'
       if not path or #path == 0 then
         return
       end
       local result = path:gsub(strip_sep_pat, ''):gsub(strip_dir_pat, '')
       if #result == 0 then
-        return '/'
+        if is_windows then
+          return path:sub(1, 2):upper()
+        else
+          return '/'
+        end
       end
       return result
     end
   end
 
   local function path_join(...)
-    local result = table.concat(vim.tbl_flatten { ... }, path_sep):gsub(path_sep .. '+', path_sep)
+    local result = table.concat(vim.tbl_flatten { ... }, path_sep()):gsub(path_sep() .. '+', path_sep())
     return result
   end
 
+  local function sanitize_path(path, escape_filename)
```

Comment:
```suggestion
  local function sanitize(path, escape_filename)
```
nit: I'm thinking it's redundant to call it `util.path.santize_path`.

Is there a reason why we can't always use `fnameescape()`?